### DISCUSSION TITLE
Deno Secure Websocket Client

### DISCUSSION THREAD

**Discussion by Ghost1928563945** [denoland/deno#11155] [Q&A] [answered]
For some resson, secure websockets do not work in deno. Insecure websockets like `ws://echo.websocket.org` work. You can open a secure websocket, but the readystate never gets set to 1.
```typescript
const websocket = new WebSocket("wss://echo.websocket.org");
websocket.addEventListener('open', (event:any) => {
    console.log(event)
});
```

**Comment by crowlKats** [comment, marked as answer]
that website has an old certificate and doesnt meet up to rusttls standards. if you listen to the error event you would get an event there.

**Reply by Ghost1928563945** [reply]
Uhh, is there a way to bypass it?

**Reply by bnoordhuis** [reply]
No. That website only supports ciphers that are deemed insecure.

**Reply by Ghost1928563945** [reply]
Thank you!
